package/kernel/mac80211/patches/ath/558-ath9k-only-mask-use_eeprom-on-of-noeeprom.patch

```diff
@@ -0,0 +1,75 @@
+ath9k: Avoid OF no-EEPROM quirks without qca,no-eeprom
+
+ath9k_of_init() function[0] was initially written on the assumption that
+if someone had an explicit ath9k OF node that "there must be something
+wrong, why would someone add an OF node if everything is fine"[1]
+(Quoting Martin Blumenstingl)
+
+"it turns out it's not that simple. with your requirements I'm now aware
+of two use-cases where the current code in ath9k_of_init() doesn't work
+without modifications"[1]
+
+The "your requirements" Martin speaks of is the result of the fact that I
```

Comment:
The mac address from MTD partition is an OpenWrt specific extension and upstream might reject the patch because of this.

I would focus on the upstream available `mtd-address` property which can be hardcoded in the dts or passed/filled via bootloader. I would propose to the change the paragraph to:

```
The "your requirements" Martin speaks of is the result of the fact that
I have a device (PowerCloud Systems CR5000) which has some kind of 
default - not unique mac address - set and requires to set the correct
MAC address via mac-address devicetree property. Or more succinctly:
```

Reply:
Thanks for the tip!  Updating...